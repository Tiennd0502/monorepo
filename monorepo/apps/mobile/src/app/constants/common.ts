export const CONGRATS_IMAGE = 'https://i.ibb.co/nkf6hFr/congrats.png';
