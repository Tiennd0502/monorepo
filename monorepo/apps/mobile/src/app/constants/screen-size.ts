import { Dimensions } from 'react-native';

export const BASE_SCREEN_WIDTH = 412;
export const BASE_SCREEN_HEIGHT = 842;

export const SCREEN_WIDTH: number = Dimensions.get('window').width;
export const SCREEN_HEIGHT: number = Dimensions.get('window').height;

export const CONTENT_WIDTH: number = SCREEN_WIDTH - 40;
