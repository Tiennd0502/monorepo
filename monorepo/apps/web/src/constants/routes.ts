export const ROUTES = {
  HOME: '/',
  VERIFY_OTP: '/verify-otp',
  SIGN_UP: '/sign-up',
  LOGIN: '/login',
};
