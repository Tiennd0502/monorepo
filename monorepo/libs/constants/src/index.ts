export * from './apis';
export * from './message';
export * from './commons';
export * from './product';
export * from './category';
export * from './validation';
