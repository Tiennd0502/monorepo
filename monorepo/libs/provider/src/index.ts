export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
