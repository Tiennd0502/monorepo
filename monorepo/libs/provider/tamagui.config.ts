import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
