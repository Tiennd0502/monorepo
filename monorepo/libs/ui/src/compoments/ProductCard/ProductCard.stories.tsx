import { Stack } from 'tamagui';
import { Meta, StoryObj } from '@storybook/react';

import ProductCard from '.';
import { PRODUCTS } from '@monorepo/mocks';

const meta: Meta<typeof ProductCard> = {
  title: 'Components/ProductCard',
  component: ProductCard,
  decorators: [
    (Story) => (
      <Stack width={180}>
        <Story />
      </Stack>
    ),
  ],
};

export default meta;

type Story = StoryObj<typeof ProductCard>;

export const Default: Story = {
  args: {
    item: PRODUCTS[2],
  },
};
