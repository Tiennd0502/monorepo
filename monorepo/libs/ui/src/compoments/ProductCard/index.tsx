import { memo } from 'react';

// Types
import { Product } from '@monorepo/types';

// Constants
import { DEFAULT_PRODUCT_IMAGE } from '@monorepo/constants';

// Components
import { Stack, Image } from 'tamagui';
import Text from '../Text';

interface ProductCardProps {
  item: Product;
  onPress: () => void;
}

const ProductCard = ({
  item: { price, name, image },
  onPress,
}: ProductCardProps) => {
  const { formatted = '' } = price || {};

  return (
    <Stack testID="product" onPress={onPress} gap="$2.5" flex={1}>
      <Image
        source={{ uri: image || DEFAULT_PRODUCT_IMAGE }}
        height="$52.5"
        borderRadius="$2.5"
        width="100%"
      />
      <Stack paddingRight={10}>
        <Text color="$textQuaternary" numberOfLines={1} ellipsizeMode="tail">
          {name}
        </Text>
        <Text color="$textBold" fontWeight="bold">
          {formatted}
        </Text>
      </Stack>
    </Stack>
  );
};

export default memo(ProductCard);
