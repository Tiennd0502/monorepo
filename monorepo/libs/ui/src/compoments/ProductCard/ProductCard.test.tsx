import {fireEvent, render, waitFor} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import { PRODUCTS } from '@monorepo/constants';
import ProductCard from '.';

const onPress = jest.fn();

const baseProps = {
  item: PRODUCTS[0],
  onPress,
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <ProductCard {...props} />
    </TamaguiProvider>
  );
};

describe('ProductCard', () => {

  it('should render ProductCard component', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('calls onClick when being clicked', () => {
    const {getByTestId} = setup();

    waitFor(() => {
      const button = getByTestId('product');
      fireEvent.press(button);
  
      expect(onPress).toHaveBeenCalled();
    })
  });
});
