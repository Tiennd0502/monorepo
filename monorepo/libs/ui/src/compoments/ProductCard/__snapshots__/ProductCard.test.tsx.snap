// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render ProductCard component 1`] = `
<View
  style={
    {
      "flex": 1,
      "gap": 10,
    }
  }
  testID="product"
>
  <Image
    source={
      {
        "uri": "https://i.ibb.co/MDpMXMG/jonny-caspari-wsv-CC6-Uy-Kjs-unsplash-1.png",
      }
    }
    style={
      {
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "height": 210,
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "paddingRight": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        {
          "color": "#606060",
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 14,
          "fontWeight": "normal",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      Black Simple Lamp
    </Text>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#303030",
          "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
          "fontSize": 14,
          "fontWeight": "bold",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      $ 12.00
    </Text>
  </View>
</View>
`;
